src/dataaccess/McodeV05SmartOnFhirDataSource.jsx

```diff
@@ -105,6 +105,13 @@ class McodeV05SmartOnFhirDataSource extends IDataSource {
                 // response is a Bundle of type searchset
                 // TODO: error handling?
                 if (response && response.entry) {
+                    // If we need to use a mapper to add profiles to the fhir, update the resources accordingly
+                    if (mapper) {
+                        const resources = response.entry.map(e => e.resource);
+                        const results = mapper.execute(resources);
+                        const wrappedResults = results.map(resource => ({ fullUrl: `urn:uuid:${resource.id}`, resource, request: { method: 'POST', url: resource.resourceType } }));
```

Comment:
I know this was in the sample code that I sent to you, but looking at it now I'm concerned about rewriting the fullUrl like this. Maybe it doesn't matter though. Making a note here mostly so I remember to look at this again on Monday

Reply:
Do you see any good way of preserving the fullUrl from the entry through this mapping process? I think the mapper will preserve order but I don't know if it's guaranteed.

When I originally wrote this, it was in the context of updating local json bundles generated by synthea, and I know that this was how the fullUrl was originally created, with resource.id being a UUID so it's globally unique. Arbitrary data coming from a server may have resource IDs that are unique among that resource type but not globally unique. So my fear is that this may produce entries with duplicate fullIUrls.